src/SmartComponents/Systems/SystemsTable.js

```diff
@@ -60,7 +60,23 @@ const SystemsTable = ({
     ]);
 
     const { systemProfile, selectedTags,
-        filter, search, page, perPage, sort } = queryParams;
+        filter: queryParamsFilter, search, page, perPage, sort
+    } = queryParams;
+    const { os: operatingSystemFilter, ...filter } = queryParamsFilter || {};
+    const osFilter = operatingSystemFilter && [{
+        osFilter: operatingSystemFilter.reduce((osFilter, os) => {
+            const [osName, osVersion] = os.split(' ');
+            const [major] = osVersion.split('.');
+
+            return {
+                ...osFilter,
+                [`${osName}-${major}`]: {
+                    ...osFilter[`${osName}-${major}`] || {},
```

Comment:
nit-pick:
```suggestion
                    ...osFilter[`${osName}-${major}`],
```

Reply:
Not falling back to `{}` could cause an error, because it the major version could be missing.

Reply:
true. I confused it with `[ ...undefined ]`